ext/objspace/objspace.c

```diff
@@ -38,10 +38,14 @@
  *  information as only a *HINT*. Especially, the size of +T_DATA+ may not be
  *  correct.
  *
- *  This method is only expected to work with C Ruby.
- *
- *  From Ruby 2.2, memsize_of(obj) returns a memory size includes
- *  sizeof(RVALUE).
+ *  This method is only expected to work with CRuby.
+ *
+ *  From Ruby 2.2, memsize_of(obj) returns a memory size that includes
+ *  the slot size occupied by the object. Prior to Ruby 3.2, this was a
+ *  fixed size (sizeof(RVALUE)). From Ruby 3.2 with Variable Width
+ *  Allocation, this is the actual slot size used (1×, 2×, 4×, 8×, or
```

Comment:
I got this information from `rb_gc_impl_objspace_init`: https://github.com/Shopify/ruby/blob/9eac1c3a888088912fcc1cbb6fc677fd1ad5683c/gc/default/default.c#L9331-L9337

Let me know if it's correct?